Userland/su.cpp

```diff
@@ -47,66 +45,34 @@ int main(int argc, char** argv)
     if (geteuid() != 0)
         fprintf(stderr, "Not running as root :(\n");
 
-    uid_t uid = 0;
-    gid_t gid = 0;
-    struct passwd* pwd = nullptr;
-    if (user) {
-        pwd = getpwnam(user);
-        if (!pwd) {
-            fprintf(stderr, "No such user: %s\n", user);
-            return 1;
-        }
-        uid = pwd->pw_uid;
-        gid = pwd->pw_gid;
-    }
-
-    if (!pwd)
-        pwd = getpwuid(0);
-
-    if (!pwd) {
+    auto account_or_error = (user) ? Core::Account::get(user) : Core::Account::get(0);
+    if (account_or_error.is_error()) {
         fprintf(stderr, "No passwd entry.\n");
         return 1;
     }
 
-    if (getuid() != 0 && pwd->pw_passwd[0] != '\0') {
+    Core::Account account = account_or_error.value();
+
+    if (getuid() != 0 && account.passwd() != "") {
         auto password = Core::get_password();
         if (password.is_error()) {
             fprintf(stderr, strerror(password.error()));
```

Comment:
Another `strerror`-based injection vulnerability! 😆

Reply:
This, too, still unsolved, though these lines are not changed by this PR, would be nice to fix nevertheless.